sprocs/check_assessment_access_rule.sql

```diff
@@ -59,86 +59,79 @@ BEGIN
         ps_course_id bigint;
         reservation reservations;
     BEGIN
-        -- only needed for Exams when we care about the date
-        EXIT schedule_access WHEN assessment_access_rule.mode IS DISTINCT FROM 'Exam';
+        -- reservation checking implies we are given a date to check
```

Comment:
Do we really want to skip PS checking when `use_date_check = FALSE`? We set `use_date_check = FALSE` when we want to generate the list of access rules to show to students (so they can see what is coming up or was in the past). I think we don't want to show them this information for linked exams?

Reply:
That's a carry over from the previous iteration of the function. I might have put that check in the wrong IF stanza when I moved things around.

The student won't see the exam at all unless they're checked in, and then it's just what the "Available Credit" and question mark popover show.

The "examId" rule could be with "startDate" and "endDate", or by itself, so those are the edge cases to think about. If there's no date associated and just the examId, a checked in student for Exam 3 would see:

![screen shot 2018-10-11 at 9 50 25 am](https://user-images.githubusercontent.com/10765199/46813732-e0ea8d80-cd3c-11e8-975e-7c7aebb295d2.png)

If there is a date range, then they'd see the whole date range (which I think you want to avoid):
![screen shot 2018-10-11 at 9 52 13 am](https://user-images.githubusercontent.com/10765199/46814055-8d2c7400-cd3d-11e8-84b7-371a0742c617.png)

If I move the `EXIT WHEN NOT use_date_check` into the ELSE part that only covers non-examId linked exam reservation checking, here's what I see:

If a date range is also config'd:
![screen shot 2018-10-11 at 9 46 11 am](https://user-images.githubusercontent.com/10765199/46813936-4b033280-cd3d-11e8-813f-75da07707e67.png)

Otherwise if it's just an `examId` with no date ranges:
![screen shot 2018-10-11 at 9 48 11 am](https://user-images.githubusercontent.com/10765199/46813962-56565e00-cd3d-11e8-880a-deb4e5b8320f.png)

I'm not sure which is preferable.

Reply:
@mwest1066 Thoughts on this?

Reply:
I mainly want to stop students seeing exams that they are not currently checked in for. With the `EXIT schedule_access WHEN NOT use_date_check` I think that a student will see a list all exams that are available to them in the CBTF, but then when they click on an individual exam they will be denied access if they aren't checked in for it. I'd prefer that they never see the exams at all.

Or am I misunderstanding what will happen here?

Just to be clear, I'm proposing that we completely delete the `EXIT schedule_access WHEN NOT use_date_check` line, not just move it around. I know this is a change from what we had before, but I think it's better?

Reply:
The studentAssessments page does an authz_assessment() call to create the list, so it only shows authorized (i.e. checked-in) exams on the assessment page. So the scope of what `use_date_check` does in that context is just what times are shown when the user already has access.

Reply:
What it looks like for the example course, student, exam with the `EXIT schedule_access WHEN NOT use_date_check` line removed:

![screen shot 2018-10-18 at 9 26 15 am](https://user-images.githubusercontent.com/10765199/47161659-17358900-d2b8-11e8-83aa-cbd2c51427a1.png)